### PR TITLE
Simplify alias declaration type in [Rules.t]

### PR DESCRIPTION
**Pull request by aalekseyev**
Instead of having three levels of "deps" declarations
(`Path.t list * Path.t list Action_builder.t`) it's sufficient to have one: `unit Action_builder.t`

Signed-off-by: Arseniy Alekseyev <aalekseyev@janestreet.com>

### CODE REVIEW

File: src/dune_engine/action_builder.mli [hunk 1]
```diff
@@ -117,6 +117,10 @@ val path_set : Path.Set.t -> unit t
     dependencies of the action produced by the action builder. *)
 val paths_matching : loc:Loc.t -> File_selector.t -> Path.Set.t t
 
+(** Like [paths_matching], but don't return the resulting set. The action
+    dependency is still registered. *)
+val paths_matching_ : loc:Loc.t -> File_selector.t -> unit t
```
Comment (jeremiedimino) [inline src/dune_engine/action_builder.mli:122]: What do you of think of `paths_matching_unit`? That's the suffix we use for `dyn_paths_unit`.

Comment (jeremiedimino) [reply src/dune_engine/action_builder.mli:122]: It's a bit odd to read though, feels like "paths that are matching unit".

**Review by jeremiedimino — APPROVED** [review]
Looks good, the new API is definitely simpler!

File: src/dune_engine/action_builder.mli (hunk 1 above)
Comment (aalekseyev) [reply src/dune_engine/action_builder.mli]: Either name seems OK to me. I changed it to `paths_matching_unit`.

**Pull request merged**